From 20ec8510d0ca2ac789b1ca9cf7b0a6dc2d02f14c Mon Sep 17 00:00:00 2001
From: Seth Forshee <seth.forshee@canonical.com>
Date: Mon, 28 Oct 2013 08:15:59 -0500
Subject: [PATCH 6/6] Makefile: Ensure _GNU_SOURCE really is always defined

When CLFAGS is set with a command argument a normal variable
assignment is not sufficient to ensure that _GNU_SOURCE will be
defined. Since this really is critical for building, use an
override directive when adding it to CFLAGS to ensure that it
really does get defined in all cases.

Signed-off-by: Seth Forshee <seth.forshee@canonical.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1d613c4..a2ba385 100644
--- a/Makefile
+++ b/Makefile
@@ -223,7 +223,7 @@ CFLAGS ?= -g -Wall
 LDFLAGS ?=
 
 # Required CFLAGS
-CFLAGS += -D_GNU_SOURCE
+override CFLAGS += -D_GNU_SOURCE
 
 ifndef NO_PTRACE
 ifneq ($(call try-cc,$(SOURCE_PTRACE),),y)
-- 
1.8.3.2

